File: leetcode/1833-maximum-ice-cream-bars/solution.cpp
class Solution {
public:
    /**
     * RUNTIME: 393 ms (37.65%)
     * MEMORY: 76.4 MB (76.68%)
    */
    int maxIceCream(vector<int>& costs, int coins) {
        sort(begin(costs), end(costs));
        for (int i = 0; i < costs.size(); ++i) {
            if (coins < costs[i]) { return i; }
            coins -= costs[i];
        }
        return costs.size();
    }
};

File: leetcode/1833-maximum-ice-cream-bars/solution.py
class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        """
        RUNTIME: 855 ms (88.48%)
        MEMORY: 28 MB (61.26%)
        """
        if sum(costs) < coins: return len(costs)
        costs.sort()
        answer = 0
        for i in costs:
            if coins < i:
                break
            coins -= i
            answer += 1
        return answer
